### PR TITLE
Update project for Xcode 12 and iOS 14

### PR DESCRIPTION
**Pull request by SergioEstevao**
This PR updates the code for Xcode 12 and iOS 14:

-  Check for PhotoLibrary permissions correctly in iOS 14.

On iOS 14 there is a new permission system for accessing the media library. 

The method to check for permission was deprecated and a new method was added: `requestAuthorizationForAccessLevel`

The answers to the new method have a new option called "PHAuthorizationStatusLimited" and it allows users to limit the media that is available to third-parties.

After a user selects this new option, the app will allow changing the media selected, after a "cold" restart.

This PR updates the code to use the new method when available. 

I'm considering if we should alert the user that they restricted their media access so only a small subset is available and direct the user to the settings to change that.

To test:
 - Start the demo app
 - Press the + button
 - Check that access to Photos permissions are requested and the option to "Select Media" works correctly.

### CODE REVIEW

**Comment by SergioEstevao** [comment]
@frosty and @loremattei do you mind give a quick look at this one?

**Review by loremattei — APPROVED** [review]
This looks good to me, but I'm not super familiar with media, so another pair of eyes would be appreciated.

File: Pod/Classes/WPPHAssetDataSource.m [hunk 1]
```diff
@@ -119,61 +119,70 @@ - (void)photoLibraryDidChange:(PHChange *)changeInstance
...
-                        return;
-                    }
-                    case (WPMediaLoadOptionsAssets): {
-                        [self loadAssetsWithSuccess:successBlock failure:failureBlock];
-                        return;
+                return;
+            }
+            case PHAuthorizationStatusNotDetermined:
+            {
+                [PHPhotoLibrary requestAuthorization:^(PHAuthorizationStatus status) {
```
Comment (frosty) [inline Pod/Classes/WPPHAssetDataSource.m]: I may not be following the code correctly, but should we be doing the iOS 14 check here again, or not? How come we don't call `requestAuthorizationForAccessLevel` again here? 🤔

**Review by frosty — APPROVED** [review]
This looks good to me, and testing works as expected – I just had one small query about the code, but I suspect I'm just not following something :)

File: Pod/Classes/WPPHAssetDataSource.m (hunk 1 above)
Comment (SergioEstevao) [reply Pod/Classes/WPPHAssetDataSource.m]: You are absolutely correct! Just updated the code.

**Review by frosty — APPROVED** [review]
Looks good, thanks for changing :)

**Pull request merged**